source/mainnet/smart-contracts/guides/unit-test-contract.rst

```diff
@@ -482,11 +482,169 @@ account:
         let amount1 = Amount::from_ccd(20);
         claim_eq!(host.get_transfers_to(receiver0), [amount0, amount1]);
 
+.. _writing_property_based_tests:
+
+Writing property-based tests
+============================
+
+Property-based testing technique allows for testing statements about your code that expected to be true for any input parameters.
+The input to such tests is generated randomly.
+Property-based testing is supported using the |QuickCheck|_ crate.
+The tests should placed in the same module as Wasm unit tests and annotated with the ``#[concordium_quickcheck]`` macro.
+The macro takes a named attribute ``num_tests`` for specifying the number of random tests to run.
+If no ``num_tests`` given, the default number is ``100``.
+The return value of the function should be a boolean value with ``true`` corresponding to the fact that the property holds.
+In ``Cargo.toml``, ``concordium-std`` should be specified as a ``dev`` dependency with the ``concordium-quickcheck`` feature enabled.
+
+.. code-block::
+
+    ...
+
+    [dev-dependencies]
+    concordium-std = { version = "5", features = ["concordium-quickcheck"] }
+
+    ...
+
+In the code snipped below, the parameters ``address`` and ``amount`` are randomly generated.
+The process of generating random input and running the test is repeated ``num_tests = 500``.
+
+.. code-block:: rust
+
+    #[concordium_cfg_test]
+    mod test {
+
+       #[concordium_quickcheck(num_tests = 500)]
+       fn some_property_test(address: Address, amount: Amount) -> bool {
+        ...
+        // Instantiate custom struct with random parameters.
+        let input = MyParameters { sender: address, payment: amount }
```

Comment:
It is a bit unclear that this isn't always needed.

Reply:
Ok, I'll add "if necessary" :)